### PR TITLE
Change (deprecated) KeyboardEvent.keyCode -> KeyboardEvent.key

### LINKED ISSUE #17553: Remove usages of deprecated property KeyboardEvent.keyCode
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

KeyboardEvent.keyCode has been deprecated for some time and AMP has a few usages.

KeyboardEvent.key has broad support and a better API.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

Unsure if switching to the new API would change bundle size.

### PR DESCRIPTION
**Pull request by nainar**
Fixes #17553

KeyboardEvent.keyCode is deprecated in favor KeyboardEvent.key.

KeyboardEvent.key has broad support and a better API.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

### CODE REVIEW

**Comment by nainar** [comment]
Done! Should I be concerned that no test failed :/

Swapped out KeyCodes to be strings instead of numbers. Used the console output here: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key to get the new values

**Comment by cvializ** [comment]
Hmm I checked and it doesn't look concerning. I think the tests didn't fail because your changes updated the property name to be correct in the tests, and the non-test values and properties matched in the mock event objects. It looks like the spec agrees with your results https://www.w3.org/TR/uievents-key/#named-key-attribute-values

File: extensions/amp-story/1.0/test/test-amp-story.js [hunk 1]
```diff
@@ -279,8 +279,8 @@ describes.realWin('amp-story', {
         });
 
     const eventObj = createEvent('keydown');
-    eventObj.keyCode = KeyCodes.RIGHT_ARROW;
-    eventObj.which = KeyCodes.RIGHT_ARROW;
+    eventObj.key = Keys.RIGHT_ARROW;
+    eventObj.which = Keys.RIGHT_ARROW;
```
Comment (cvializ) [inline extensions/amp-story/1.0/test/test-amp-story.js]: I am not sure why this code mocks the `which` property, but the spec defines it to be a specifically numerical value. https://www.w3.org/TR/uievents/#dom-keyboardevent-which The test code should either remove `which` if it is not needed, or if it is needed, the value should conform to the spec

Comment (nainar) [reply extensions/amp-story/1.0/test/test-amp-story.js]: Turns out `keyCode` and `which` might differ on some browsers. https://unixpapa.com/js/key.html 

I thought it safest to keep the code check to numeric here. 
Fixed in the latest patch.

**Comment by nainar** [comment]
Made the changes in the other spots. Thanks for the review!

**Pull request merged**